gcm_filters/filter.py

```diff
@@ -411,7 +411,19 @@ def plot_shape(self, ax=None):
         ax.grid(True)
         ax.legend()
 
-    def apply(self, field, dims):
+    def apply(self, field_or_dataset, dims):
+        """Filter a field or xarray dataset with scalar Laplacian across the
+        dimensions specified by dims."""
+        if isinstance(field_or_dataset, xr.Dataset):
+            filtered = field_or_dataset.copy(deep=True)
+            for key, field in filtered.variables.items():
+                if all(dim in field.dims for dim in dims):
+                    filtered[key] = self.apply_to_field(field, dims=dims)
```

Comment:
Should we add a warning here if `field` does not have the correct `dims`? (Just a suggestion - I'm not feeling strongly about it.) Otherwise, there is no obvious way for the user to see if fields in the dataset got actually filtered or not. For instance, the user may expect all fields in the dataset to be filtered but forgot to rename the spatial dimensions to `dims` for some of the fields. 

For context, if the user applies the `apply` method to a field (rather than a dataset) that does not have the dimensions `dims`, then an error message will be raised.

Reply:
(Added a commit to that effect, with some tests!)

Reply:
Great, I think the added warning is useful. Thanks for your work!